pkgs/by-name/sp/sparrow-wifi/package.nix

```
@@ -0,0 +1,179 @@
+{
+  lib,
+  python3,
+  fetchFromGitHub,
+  fetchpatch,
+  qt5,
+  gpsd,
+  usbutils,
+  wirelesstools,
+  iw,
+  bluez,
+  ubertooth,
+  aircrack-ng,
+  john,
+  imagemagick,
+  makeDesktopItem,
+  writeShellScript,
+  copyDesktopItems,
+}:
+let
+  sparrow-wifi-desktop = makeDesktopItem {
+    name = "sparrow-wifi";
+    desktopName = "Sparrow WiFi";
+    exec = "sparrow-wifi-pkexec";
+    comment = meta.description;
+    icon = "sparrow_wifi_icon";
+    categories = [ "Network" ];
+    type = "Application";
+    terminal = false;
+  };
+
+  sparrow-wifi-pkexec = writeShellScript "sparrow-wifi-pkexec" ''
+    # pkexec wrapper that passes required variables for X11 and Wayland
+    #
+    # org.freedesktop.policykit.exec.allow_gui only passes variables for X11
+    # so it doesnt work for Wayland
+
+    pkexec env \
+      DISPLAY=$DISPLAY \
+      XAUTHORITY=$XAUTHORITY \
+      WAYLAND_DISPLAY=$WAYLAND_DISPLAY \
+      XDG_RUNTIME_DIR=$XDG_RUNTIME_DIR \
+      XDG_SESSION_TYPE=$XDG_SESSION_TYPE \
+      QT_QPA_PLATFORM=$QT_QPA_PLATFORM \
+      @out@/bin/sparrow-wifi "$@"
+  '';
+
+  meta = {
+    homepage = "https://github.com/ghostop14/sparrow-wifi";
+    description = "Next-Gen GUI-based WiFi and Bluetooth Analyzer for Linux";
+    mainProgram = "sparrow-wifi";
+    license = lib.licenses.gpl3;
+    maintainers = with lib.maintainers; [ dsuetin ];
+    platforms = lib.platforms.linux;
+  };
+in
+python3.pkgs.buildPythonApplication {
+  pname = "sparrow-wifi";
+  version = "0-unstable-2024-07-25";
+  format = "other";
```

```suggestion
  pyproject = false;
```
Is preferred and has the same effect if I remember correctly.

---

Done, I thought that it will default to setuptools, but `false` does indeed default to `other`.